### PR TITLE
Fix mypy errors attributed to `pytorch_lightning.profilers.pytorch`

### PR DESCRIPTION
**Pull request by krishnakalyan3**
Part of https://github.com/Lightning-AI/lightning/issues/13445

### CODE REVIEW

**Review by krishnakalyan3 — COMMENTED** [review]
fix conflicts

**Review by krishnakalyan3 — COMMENTED** [review]
fix conflicts

**Review by krishnakalyan3 — COMMENTED** [review]
commit

**Review by rohitgr7 — APPROVED** [review]

**Comment by krishnakalyan3** [comment]
Thank you @rohitgr7. I will look at the changes.

File: src/pytorch_lightning/profilers/pytorch.py [hunk 1]
```diff
@@ -469,14 +478,16 @@ def summary(self) -> str:
...
     def _create_profilers(self) -> None:
-        if self._emit_nvtx:
-            self._parent_profiler = self._create_profiler(torch.cuda.profiler.profile)
-            self.profiler = self._create_profiler(torch.autograd.profiler.emit_nvtx)
-        else:
-            self._parent_profiler = None
-            self.profiler = self._create_profiler(
-                torch.profiler.profile if _KINETO_AVAILABLE else torch.autograd.profiler.profile
-            )
+        if self.profiler is None:
```
Comment (awaelchli) [inline src/pytorch_lightning/profilers/pytorch.py:481]: Nit: the if-clause could be inverted with an early return to avoid indentation.

**Review by awaelchli — APPROVED** [review]

**Review by Borda — APPROVED** [review]

File: src/pytorch_lightning/profilers/pytorch.py [hunk 2]
```diff
@@ -469,14 +478,16 @@ def summary(self) -> str:
...
+            if self._emit_nvtx:
+                if self._parent_profiler is None:
+                    self._parent_profiler = torch.cuda.profiler.profile()
+                self.profiler = self._create_profiler(torch.autograd.profiler.emit_nvtx)
+            else:
+                self._parent_profiler = None
+                self.profiler = self._create_profiler(
+                    torch.profiler.profile if _KINETO_AVAILABLE else torch.autograd.profiler.profile
+                )
```
Comment (rohitgr7) [inline src/pytorch_lightning/profilers/pytorch.py:490]:
```suggestion
        if self.profiler is not None:
            return

        if self._emit_nvtx:
            if self._parent_profiler is None:
                self._parent_profiler = torch.cuda.profiler.profile()
            self.profiler = self._create_profiler(torch.autograd.profiler.emit_nvtx)
        else:
            self._parent_profiler = None
            self.profiler = self._create_profiler(
                torch.profiler.profile if _KINETO_AVAILABLE else torch.autograd.profiler.profile
            )
```

**Pull request merged**